src/lib/y2network/autoinst_profile/section_with_attributes.rb

```diff
@@ -0,0 +1,189 @@
+        # contain the relevant information. Attributes are set to nil for
+        # missing keys and for blank values.
+        #
+        # @param hash   [Hash] content of the corresponding section of the profile.
+        #   Each element of the hash corresponds to one of the attributes
+        #   defined in the section.
+        # @param parent [#parent,#section_name] parent section
+        # @return [SectionWithAttributes]
+        def new_from_hashes(hash, parent = nil)
+          result = new(parent)
+          result.init_from_hashes(hash)
+          result
+        end
+
+      protected
+
+        # Macro used in the subclasses to define accessors for all the
+        # attributes defined by {.attributes}
+        def define_attr_accessors
+          attributes.each do |attrib|
+            attr_accessor attrib[:name]
+          end
+        end
+      end
+
+      # This value only makes sense when {.new_from_hashes} is used.
+      #
+      # @return [#parent,#section_name] Parent section
+      attr_reader :parent
+
+      # Constructor
+      #
+      # @param parent [SectionWithAttributes] Parent section
+      def initialize(parent = nil)
+        @parent = parent
+      end
+
+      # Method used by {.new_from_hashes} to populate the attributes.
+      #
+      # By default, it simply assigns the non-empty hash values to the
+      # corresponding attributes, logging unknown keys. The subclass is expected
+      # to refine this behavior if needed.
+      #
+      # @param hash [Hash] see {.new_from_hashes}
+      def init_from_hashes(hash)
+        init_scalars_from_hash(hash)
+      end
+
+      # Content of the section in the format used by the AutoYaST modules
+      # (nested arrays and hashes).
+      #
+      # @return [Hash] each element of the hash corresponds to one of the
+      #     attributes defined in the section. Blank attributes are not
+      #     included.
+      def to_hashes
+        attributes.each_with_object({}) do |attrib, result|
+          value = attrib_value(attrib)
+          next if attrib_skip?(value)
+
+          key = attrib_key(attrib)
+          result[key] = value
+        end
+      end
+
+      # Returns the section name
+      #
+      # In some cases, the section name does not match with the XML name
+      # and this method should be redefined.
+      #
+      # @example
+      #   section = PartitioningSection.new
+      #   section.section_name #=> "partitioning"
+      #
+      # @return [String] Section name
+      def section_name
+        klass_name = self.class.name.split("::").last
+        klass_name
+          .gsub(/([a-z])([A-Z])/, "\\1_\\2").downcase
+          .chomp("_section")
+      end
+
+    protected
+
+      def attributes
+        self.class.attributes
+      end
+
+      # Whether an attribute must be skipped during import/export.
+      #
+      # @return [Boolean] true is the value is blank
+      def attrib_skip?(value)
```

Comment:
I have to say that this accronym is a bit strange. I expect `attr` or full `attribute` this is quite long and still not full.